File: leetcode_724.py
from typing import List


class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        lastLeft = 0
        left = 0
        right = sum(nums)

        foundPos = -1
        for i, x in enumerate(nums):
            left += lastLeft
            right -= x

            if left == right:
                foundPos = i
                break
            
            lastLeft = x
        
        return foundPos

sol = Solution()

def test_case_001():
    nums = [1,7,3,6,5,6] # output 3
    assert sol.pivotIndex(nums) == 3

def test_case_002():
    nums = [1,2,3] # output -1
    assert sol.pivotIndex(nums) == -1
    
def test_case_003():
    nums = [2,1,-1] # output 0
    assert sol.pivotIndex(nums) == 0

File: JavaScript/leetcode_724.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var pivotIndex = function (nums) {
  // let sum = 0;
  let leftSideTotal = 0;
  let rightSideTotal = 0;
  nums.forEach((n) => {
    rightSideTotal += n;
  });

  let i = 0;
  for (n of nums) {
    rightSideTotal -= n;
    if (leftSideTotal === rightSideTotal) return i;
    leftSideTotal += n;
    i++;
  }

  return -1;
};


// nums = [1, 7, 3, 6, 5, 6] // expected output : 3
// nums = [1, 2, 3]; // expected output : -1
// nums = [1, 1]; // expected output : -1
// nums = [1, 1, 1]; // expected output : 1
// nums = [1, 0, 1]; // expected output : 1
nums = [0, 1]; // expected output : 1
console.log(pivotIndex(nums));